llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -2473,6 +2473,16 @@ Instruction *InstCombinerImpl::visitURem(BinaryOperator &I) {
     }
   }
 
+  Value *A;
+  const APInt *Op1Cst, *BCst;
+  // urem(urem(A, BCst), Op1Cst) -> urem(A, Op1Cst)
+  // iff urem(BCst, Op1Cst) == 0
+  if (match(Op0, m_URem(m_Value(A), m_APInt(BCst))) &&
```

Comment:
Looks like we need a one-use check on the inner urem. Otherwise we often end up breaking up div+rem patterns, and also performing the urem on a wider type.

Reply:
ok, can you rerun the benchmark? (I am not sure whether I need rights for that)

Reply:
It looks like with this change the fold never triggers anymore...

Reply:
It seems so that it's always this div+rem pair in the pattern posted in the original issue. (actually found only once)

https://github.com/dtcxzyw/llvm-opt-benchmark/blob/main/bench/ffmpeg/optimized/ac3dec_fixed.ll#L7580